twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php

```diff
@@ -221,20 +223,27 @@ private function triggerChallenges($username)
      */
     public function getTemplate(IUser $user)
     {
-        $messages = [];
-        if ($this->getAppValue('triggerchallenges', '') === '1') {
-            try {
-                $messages = $this->triggerChallenges($user->getUID());
-            } catch (TriggerChallengesException $e) {
-                $messages = [$e->getMessage()];
-            }
-        }
+
+    	$transactionId = $this->session->get("pi_transaction_id");
+
+    	if (!$transactionId) {
+			$messages = [];
+			if ($this->getAppValue('triggerchallenges', '') === '1') {
+				try {
+					$messages = $this->triggerChallenges($user->getUID());
+					$this->session->set("pi_message", $messages);
+				} catch (TriggerChallengesException $e) {
+					$messages = [$e->getMessage()];
+				}
+			}
+		} else {
+    		$messages = $this->session->get("pi_message");
```

Comment:
If I have a "normal" second factor like a smartphone app or yubikey, that is *no* challenge response token and I use no triggerchallenge, would we then set the message to an emtpy string here?

Reply:
Yes, but it is not displayed. If we do not sent a message, only the default one will be shown.
![Bildschirmfoto vom 2019-05-31 08-23-15](https://user-images.githubusercontent.com/35998229/58686017-d76e6e00-837d-11e9-9fd8-c1e3566a4f63.png)